packages/react-composites/tests/browser/chat/hermetic/ImageOverlay.test.ts

```diff
@@ -17,29 +17,74 @@ test.describe('ImageOverlay tests', () => {
+    expect(await stableScreenshot(page)).toMatchSnapshot(`image-overlay-modal.png`);
   });
 
   test('ImageOverlay should show broken image icon with alt text when url is a broken link', async ({
     page,
     serverUrl
   }) => {
-    try {
-      await page.goto(
-        buildUrlForChatAppUsingFakeAdapter(serverUrl, {
-          localParticipant: TEST_PARTICIPANTS[1],
-          remoteParticipants: [TEST_PARTICIPANTS[0], TEST_PARTICIPANTS[2]],
-          localParticipantPosition: 1,
-          sendRemoteInlineImageMessage: true,
-          inlineImageUrl: 'images/inlineImage-broken.png'
-        })
-      );
-    } catch (error) {
-      // This is expected to fail because the image url is a broken link,
-      // we want to swallow this error to prevent it from logging to the console
-      await page.locator(dataUiId('SomeImageId1')).click();
-      expect(await stableScreenshot(page)).toMatchSnapshot(`inline-image-overlay-modal-broken-link.png`);
-    }
+    // Mock the api call before navigating
+    await page.route(serverUrl + '/images/inlineImageExample1.png', async (route) => {
+      try {
+        await route.continue();
+      } catch (error) {
+        console.error('Failed at continue on route, Error: ', error);
+      }
+    });
+
+    await page.route(serverUrl + '/images/inlineImageExample1-fullSize.png', async (route) => {
+      try {
+        await route.fulfill({ status: 300, contentType: 'text/html' });
+      } catch (error) {
+        console.error('Failed at fulfill route, Error: ', error);
+      }
+    });
+
+    await page.goto(
+      buildUrlForChatAppUsingFakeAdapter(serverUrl, {
+        localParticipant: TEST_PARTICIPANTS[1],
+        remoteParticipants: [TEST_PARTICIPANTS[0], TEST_PARTICIPANTS[2]],
+        localParticipantPosition: 1,
+        sendRemoteInlineImageMessage: true
+      })
+    );
+
+    await page.locator(dataUiId('SomeImageId1')).click();
+    expect(await stableScreenshot(page)).toMatchSnapshot(`image-overlay-modal-broken-link.png`);
+  });
+
+  test('ImageOverlay should show a placeholder box when loading', async ({ page, serverUrl }) => {
+    // Mock the api call before navigating
+    await page.route(serverUrl + '/images/inlineImageExample1-fullSize.png', async (route) => {
+      setTimeout(async () => {
+        try {
+          await route.continue();
+        } catch (error) {
+          console.error('Failed at continue on route, Error: ', error);
+        }
+      }, 3000);
+    });
+
+    await page.goto(
+      buildUrlForChatAppUsingFakeAdapter(serverUrl, {
+        localParticipant: TEST_PARTICIPANTS[1],
+        remoteParticipants: [TEST_PARTICIPANTS[0], TEST_PARTICIPANTS[2]],
+        localParticipantPosition: 1,
+        sendRemoteInlineImageMessage: true
+      })
+    );
+
+    await page.locator(dataUiId('SomeImageId1')).click();
+
+    expect(await stableScreenshot(page, { stubMessageTimestamps: true })).toMatchSnapshot(
```

Comment:
Is it possible that this step may be lead to unstable tests? (when an image is shown faster than expected and there will not be loading state?)